sprintMission4_/src/services/productService.js

```diff
@@ -0,0 +1,115 @@
+import prisma from '../lib/prisma.js'
+
+async function register(authorId, data){
+  try{
+    const { name, description, price, tags } = data;
+    const tagsData = (Array.isArray(tags) && tags.length > 0 ) ? tags : null;
+    const newProduct = await prisma.product.create({
+      data: {
+        name,
+        description,
+        price,
+        tags: tagsData,
+        authorId,
+      }
+    })
+    return newProduct;
+  } catch(error){
+    throw error;
+  }
+}
+
+async function update(productId, data){
+  try{
+    const updatedProduct = await prisma.product.update({
+      where: { id: productId },
+      data: { data }
+    });
```

Comment:
실제 업데이트되는 필드들이 data 프로퍼티 안에 중첩되어 저장되는 구조입니다.
Prisma는 data객체의 바로 하위  프로퍼티를 데이터베이스 필드로 인식하기 때문에 이렇게되면,
data라는 프로퍼티안에 JSON형태로 데이터가 입력되게 됩니다. 에러가 발생하겠네요!

```suggestion
    const updatedProduct = await prisma.product.update({
      where: { id: productId },
      data,
    });
```